helm-charts/dnsbl-exporter/values.yaml

```diff
@@ -0,0 +1,339 @@
+##
+extraEnvVarsSecret: ""
+
+## Configure extra options for Application containers' liveness, readiness and startup probes
+## ref: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#configure-probes
+## @param startupProbe.enabled Enable startupProbe
+## @param startupProbe.initialDelaySeconds Initial delay seconds for startupProbe
+## @param startupProbe.periodSeconds Period seconds for startupProbe
+## @param startupProbe.timeoutSeconds Timeout seconds for startupProbe
+## @param startupProbe.failureThreshold Failure threshold for startupProbe
+## @param startupProbe.successThreshold Success threshold for startupProbe
+##
+startupProbe:
+  enabled: false
+  initialDelaySeconds: 120
+  periodSeconds: 10
+  timeoutSeconds: 5
+  failureThreshold: 6
+  successThreshold: 1
+## @param livenessProbe.enabled Enable livenessProbe
+## @param livenessProbe.initialDelaySeconds Initial delay seconds for livenessProbe
+## @param livenessProbe.periodSeconds Period seconds for livenessProbe
+## @param livenessProbe.timeoutSeconds Timeout seconds for livenessProbe
+## @param livenessProbe.failureThreshold Failure threshold for livenessProbe
+## @param livenessProbe.successThreshold Success threshold for livenessProbe
+##
+livenessProbe:
+  enabled: true
+  initialDelaySeconds: 120
+  periodSeconds: 10
+  timeoutSeconds: 5
+  failureThreshold: 6
+  successThreshold: 1
+## @param readinessProbe.enabled Enable readinessProbe
+## @param readinessProbe.initialDelaySeconds Initial delay seconds for readinessProbe
+## @param readinessProbe.periodSeconds Period seconds for readinessProbe
+## @param readinessProbe.timeoutSeconds Timeout seconds for readinessProbe
+## @param readinessProbe.failureThreshold Failure threshold for readinessProbe
+## @param readinessProbe.successThreshold Success threshold for readinessProbe
+##
+readinessProbe:
+  enabled: true
+  initialDelaySeconds: 30
+  periodSeconds: 5
+  timeoutSeconds: 3
+  failureThreshold: 6
+  successThreshold: 1
+## @param customLivenessProbe Custom livenessProbe that overrides the default one
+##
+customLivenessProbe: {}
+## @param customReadinessProbe Custom readinessProbe that overrides the default one
+#
+customReadinessProbe: {}
+## Custom startupProbe block
+customStartupProbe: {}
+
+image:
+  registry: ghcr.io
+  repository: luzilla/dnsbl_exporter
+  tag: v0.6.0
+  pullPolicy: IfNotPresent
+  ## Optionally specify an array of imagePullSecrets.
+  ## Secrets must be manually created in the namespace.
+  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
+  ## e.g:
+  ## pullSecrets:
+  ##   - myRegistryKeySecretName
+  ##
+  pullSecrets: []
+
+configuration:
+  rbls:
+    # 0SPAM
+    - bl.0spam.org
+    - rbl.0spam.org
+    # NIXSPAM
+    - ix.dnsbl.manitu.net
+    # Spamhaus Zen
+    - zen.spamhaus.org
+  targets:
+    - smtp.fastmail.com
+  resolver: "1.1.1.1:53"
```

Comment:
Typically cloudflare doesn't work well:
https://www.spamhaus.com/resource-center/if-you-query-spamhaus-projects-dnsbls-via-cloudflares-dns-move-to-the-free-data-query-service/

Reply:
Personally I have my own DNS server in my values override file. What you recommend to put by default?

Reply:
I mean, if you want to have a chart that brings what people "need", then a pod with unbound would be a good idea?

Reply:
@till actually I would use unbound as a sidecar. Or as a separate service (btw bitnami provides that as well). Not everyone needs a Swiss knife all the times. And additional software inside the container could provide additional vulnerabilities. So better to keep the services isolated.

Reply:
Sorry, this is a mis-understanding.

My suggestion is to add another container (not sidecar) with unbound into the same pod (not container).

Reply:
@till so “add another container with unbound in the same port” means that the sidecar should be used.
Technically pod runs from 1 to infinite containers sharing some resources like the network, etc.
The sidecar is what is definitely not described by the author of the helm chart, but could be easily added by a helm chart customer of the customer needs extra functionality.

Reply:
So either we write the YAML or the customer.
I think we should care about the functionality of the exporter, but may create an additional readme how to add unbound feature with an additional K8S-service and sidecars/deployments